CAIPs/caip-x.md

```diff
@@ -0,0 +1,339 @@
+---
+caip: XX
+title: JSON-RPC Provider Lifecycle Methods for Session Management
+author: [Alex Donesky] (@adonesky1)
+discussions-to: TBD
+status: Draft
+type: Standard
+created: 2024-06-07
+requires: 25, 217
+---
+
+## Simple Summary
+
+CAIP-XX introduces new "lifecycle" methods and a new event for managing permissions within an existing CAIP-25 session, allowing for addition, revocation, and retrieval of permissions. These methods provide an alternative to session management via `sessionId`s, allowing `sessionId`s to be optional for CAIP-25.
+
+## Abstract
+
+This proposal aims to extend the CAIP-25 standard by defining new JSON-RPC methods for managing the lifecycle of permissions within a session. These methods allow dapps and wallets to dynamically adjust permissions, providing more granular control and better user experience. Additionally, it allows for session management without mandatory sessionIds, offering more flexibility in handling sessions.
+
+## Motivation
+
+The motivation behind this proposal is to enhance the flexibility of CAIP-25 by enabling management of session permissions without sessionIds which don't map well to extension based wallet's dapp connections and could therefore add constraints and burdens to existing flows. The proposed methods provide an intuitive way to add, revoke, and retrieve permissions within an existing session, simplifying the management of session lifecycles.
+
+### Use Case Scenarios
+
+1. **Dapp Initiated Adding Permissions To an Existing Session:**
+
+   - **Current Method:** Call `provider_authorize` again with an existing session identifier and new scopes/methods to add. This is actually somewhat ambiguous in CAIP-25 where it’s unclear if incremental permissions adds should include the full object or just scopes to add: "This object gets updated, extended, closed, etc. by successive calls and notifications, each tagged by this identifier."
```

Comment:
full object was intended but maybe not explicit enough in that sentence!

Reply:
Full object is ok but would be nice to have it be additive by default IMO?

Reply:
i'm not against it, i just need (as ever!) to see the whole thing to compare to status quo, security considerations and all 😄